src/data/markdown/docs/05 Examples/02 Tutorials/01 Get started with k6/400 Reuse and re-run tests.md

```diff
@@ -0,0 +1,418 @@
+
+
+
+## Mix and match logic
+
+With modularized configuration and logic, you can mix and match logic.
+An easy way to configure this is through [environment variables](/using-k6/environment-variables).
+
+Change `main.js` and `config.js` so that it:
+- _By default_ runs a smoke test with 5 iterations
+- With the right enviroment variable value, runs a breaking test
+
+To do this, follow these steps:
+
+1. Add the workload settings for configuring the smoke test to `config.js`:
+
+  <CodeGroup labels={["config.js"]} lineNumbers={[false]} showCopyButton={[true]} heightTogglers={[true]}>
+
+  ```javascript
+  export const smokeWorkload = {
+    executor: 'shared-iterations',
+    iterations: 5,
+    vus:1
+  } 
+
+  export const thresholdsSettings = { 
+    http_req_failed: [{ threshold: "rate<0.01", abortOnFail: true }],
+    http_req_duration: ["p(99)<1000"],
+  }
+
+  export const breakingWorkload = {
+    executor: 'ramping-vus',
+    stages: [
+      { duration: '10s', target: 20 },
+      { duration: '50s', target: 20 },
+      { duration: '50s', target: 40 },
+      { duration: '50s', target: 60 },
+      { duration: '50s', target: 80 },
+      { duration: '50s', target: 100 },
+      { duration: '50s', target: 120 },
+      { duration: '50s', target: 140 },
+      //....
+    ],
+  } 
+  ```
+
+  </CodeGroup>
+
+1. Edit `main.js` to choose the workload settings depending on the `WORKLOAD` environment variable. For example:
+   
+  <CodeGroup labels={["main.js"]} lineNumbers={[false]} showCopyButton={[true]} heightTogglers={[true]}>
+
+  ```javascript
+
+  import { coinflip } from "./coinflip.js";
+  import { contacts } from "./contacts.js";
+  import { thresholdsSettings, breakingWorkload, smokeWorkload } from "./config.js";
+
+  export const options = {
+    scenarios: {
+      my_scenario: __ENV.WORKLOAD === 'breaking' ? breakingWorkload : smokeWorkload
+    },
+    thresholds: thresholdsSettings
+  };
+
+  const baseUrl = "https://test.k6.io";
+
+  export default function () {
+    contacts(baseUrl);
+    coinflip(baseUrl);
+  }
+
+  ```
+  </CodeGroup>
+  
+1. Run the script with and without the `-e` flag.
+
+   - What happens when you run `k6 run main.js`?
+   - What about `k6 run main.js -e WORKLOAD=breaking`?
+
+This was a simple example to showcase how you can modularize a test. 
+As your test suite grows and more people are involved in performance testing, your modularization strategy becomes essential to building and maintaining an efficient testing suite.
+  
+## Next steps
+
+Now you've seen examples to write tests, assert for performance, filter results, and modularize scripts.
+Notice how the tests progress in complexity: from single endpoints to holistic tests, from small loads to large loads, and from single-tests to reusable modules. These progressions are typical in testing, with the next step being to automate. It might be impractical to automate a tutorial, but if you are interested,
```

Comment:
```suggestion
Notice how the tests progress in complexity: from single endpoints to holistic tests, from small to large loads, and from single tests to reusable modules. These progressions are typical in testing, with the next step being to automate. It might be impractical to automate a tutorial, but if you are interested,
```